### ISSUE TITLE
GUI: cannot insert after last word

### ISSUE THREAD

**Issue by heatherleaf** [MUSTE-Project/MULLE#64]
I cannot insert anything after the last word. In Exemplum Eng->Swe I should get this menu:

```
$ muste cli -G exemplum/Exemplum -L ExemplumSwe "en god kung läser en bok"
(...)
{6-6}: en god kung läser en bok [ ]
en god kung läser en bok [ Johan ]
en god kung läser en bok [ Paris ]
en god kung läser en bok [ idag ]
en god kung läser en bok [ nu ]
en god kung läser en bok [ redan ]
```

The menu is indeed in the client, `DATA.trg.menu[5]`, but when I click after the last word "bok", there's an exception in the JS console:
```
uncaught exception: No menu found. Probably because the user clicked a space between words, this is still not supported.
```

**Comment by fredefox** [comment]
Fixed in ab620f81900c982936fb59e316ee89a9c38224dc

**Comment by fredefox** [comment]
Hang on!

**Comment by fredefox** [comment]
The items you want to have included are included after 4b7e5151b600204f7de93e6bd28e4fa9cc1ff98f but I'm not sure that this is correct. Maybe it'll start giving you suggestions you weren't expecting. Are the intervals supposed to be inclusive? I think that is counter-intuitive but that commit assumes that they are.

**Comment by heatherleaf** [comment]
There's something wrong now, hopefully just replacing < by ≤ 

Here I clicked before "many" and it shows the correct menu (`{0-0}`), but "many" shouldn't be highlighted:
![image](https://user-images.githubusercontent.com/1324196/46411664-ae2d0d80-c71c-11e8-97dd-2b7b4f47eae8.png)

Here I clicked "kings" and it shows the insertion menu (`{1-1}`):
![image](https://user-images.githubusercontent.com/1324196/46411683-b08f6780-c71c-11e8-8489-aa544260f152.png)

Here I clicked after "Paris", and it shows the {3-4} menu:
![image](https://user-images.githubusercontent.com/1324196/46411686-b38a5800-c71c-11e8-8893-285fef5dd61f.png)

Here are the menus according to the CLI:
```
$ muste cli -G exemplum/Exemplum -L ExemplumEng --max-prune-depth 4 --max-adjtree-depth 4 --print-nodes " many kings love Paris "

{0-0}: [ ] many kings love Paris                             [ ] many_Det king_N love_V paris_PN
[ already ] many kings love Paris                            [ already_Adv ] many_Det king_N love_V paris_PN
[ now ] many kings love Paris                                [ now_Adv ] many_Det king_N love_V paris_PN
[ today ] many kings love Paris                              [ today_Adv ] many_Det king_N love_V paris_PN

{1-1}: many [ ] kings love Paris                             many_Det [ ] king_N love_V paris_PN
many [ bad ] kings love Paris                                many_Det [ bad_A ] king_N love_V paris_PN
many [ big ] kings love Paris                                many_Det [ big_A ] king_N love_V paris_PN
(...)

{2-2}: many kings [ ] love Paris                             many_Det king_N [ ] love_V paris_PN
many kings [ John ] love Paris                               many_Det king_N [ john_PN ] love_V paris_PN
many kings [ Paris ] love Paris                              many_Det king_N [ paris_PN ] love_V paris_PN
(...)

{4-4}: many kings love Paris [ ]                             many_Det king_N love_V paris_PN [ ]
many kings love Paris [ already ]                            many_Det king_N love_V paris_PN [ already_Adv ]
many kings love Paris [ now ]                                many_Det king_N love_V paris_PN [ now_Adv ]
many kings love Paris [ today ]                              many_Det king_N love_V paris_PN [ today_Adv ]

{3-3,2-3}: many kings [ love [ ] ] Paris                     many_Det king_N [ love_V [ ] ] paris_PN
many kings [ switch [ on ] ] Paris                           many_Det king_N [ switch_on_V [ switch_on_V ] ] paris_PN

{0-1}: [ many ] kings love Paris                             [ many_Det ] king_N love_V paris_PN
[ ] kings love Paris                                         [ ] king_N love_V paris_PN
[ a ] king loves Paris                                       [ aSg_Det ] king_N love_V paris_PN
(...)

{1-2}: many [ kings ] love Paris                             many_Det [ king_N ] love_V paris_PN
many [ books ] love Paris                                    many_Det [ book_N ] love_V paris_PN
many [ boys ] love Paris                                     many_Det [ boy_N ] love_V paris_PN
(...)

{2-3}: many kings [ love ] Paris                             many_Det king_N [ love_V ] paris_PN
many kings [ be ] Paris                                      many_Det king_N [ copula_V ] paris_PN
many kings [ break ] Paris                                   many_Det king_N [ break_V ] paris_PN
(...)

{3-4}: many kings love [ Paris ]                             many_Det king_N love_V [ paris_PN ]
many kings love [ ]                                          many_Det king_N love_V [ ]
many kings love [ John ]                                     many_Det king_N love_V [ john_PN ]
(...)

{1-2,0-2}: [ many [ kings ] ] love Paris                     [ many_Det [ king_N ] ] love_V paris_PN
[ John [ ] ] loves Paris                                     [ john_PN [ ] ] love_V paris_PN
[ Paris [ ] ] loves Paris                                    [ paris_PN [ ] ] love_V paris_PN
(...)
```

**Comment by heatherleaf** [comment]
Oh, please pull my latest changes before you do your stuff - some things might have changed...

**Comment by heatherleaf** [comment]
> I think the problem is that I can't determine what is selected in a consistent way because the "selection" behaves differently for spaces and words.

How about this?

Pseudo-client-code for determining if a word `1≤w≤n` is clickable:
- `w` is clickable if there is a menu `selection` such that there is some `(j,k)` in `selection` such that` j<w<=k`

Pseudo-client-code for determining if a space `0≤s≤n` is clickable:
- `s` is clickable if there is a `selection` such that there is some `(j,k)` in `selection` such that `j<w<=k`

Pseude-client-code for determining which words/spaces to strikeout (set class ".strike"):
- assume `selection = {(j',k'),(j'',k''),...}`
- for every word `1≤w≤n`:
  - if there is `(j,k)` in `selection` such that `j<w<=k`: strike word `w`
- for every space `0≤s≤n`:
  - if `(s,s)` is in `selection`: strike space `s`
  - if there is `(j,k)` in `selection` such that `j<s<k`: strike space `s`

**Comment by heatherleaf** [comment]
There is a difference between clickable and strikeout. The last bullet point says that some spaces can be striked out but it's not possible to click them to get the menu. Here's a hypothetical example:

    {0-0,2-5}: [ ] en flicka [ springer och hoppar ] 
    [ idag [ springer och hoppar ] ] en flicka
    [ nu [ springer och hoppar ] ] en flicka

Here it should be possible to click before "en" (insertion {0-0}), and on the words "springer", "och", "hoppar". But it should not be possible to click the space between "springer" and "och", because we're not inserting anything there. I.e., the following should be clickable to get the menu:

    []  en  flicka  [springer]  [och]  [hoppar]

But when the user has clicked {0-0} or {2-5} to show the menu, the spaces between "springer", "och" and "hoppar" should be striked out. I.e., the following should be striked:

    []  en  flicka  [springer] [] [och] [] [hoppar]

**Comment by fredefox** [comment]
I just reverted the change I talked about reverting. Since you said you have also made some changes, could I get you to give me a failing case again.

I also fixed an issue that you may not have noticed, but it was not possible to cycle menus for empty spaces.

**Comment by heatherleaf** [comment]
closing this in favour of a more general bug #66

**Comment by heatherleaf** [comment]
reopening since #66 now works

**Comment by fredefox** [comment]
This is fixed in 020ad15.

I've also made a new release onto develop. [Version 0.2.7.0](https://github.com/MUSTE-Project/MULLE/releases/tag/0.2.7.0).

**Comment by heatherleaf** [comment]
This doesn't work anymore, so reopening.

The "Eng -> Swe" example lesson has these two menus (for the top resp. bottom sentence):

```
{4-4}: many kings love Paris []
many kings love Paris [already]
many kings love Paris [now]
many kings love Paris [today]
many kings love Paris [and something]
many kings love Paris [in something]
many kings love Paris [on something]
many kings love Paris [or something]
many kings love Paris [to something]

{6-6}: en god kung läser en bok []
en god kung läser en bok [Johan]
en god kung läser en bok [Paris]
en god kung läser en bok [idag]
en god kung läser en bok [nu]
en god kung läser en bok [redan]
en god kung läser en bok [eller något]
en god kung läser en bok [i något]
en god kung läser en bok [och något]
en god kung läser en bok [på något]
en god kung läser en bok [till något]
```

But the gui doesn't react when clicking after "Paris" or after "bok".

**Comment by fredefox** [comment]
Reopening by request of @heatherleaf

**Comment by fredefox** [comment]
This should be fixed in 003d353. Can you have a look?

**Comment by fredefox** [comment]
This should be fixed in 003d353. Can you have a look?

**Comment by fredefox** [comment]
This should be fixed in 003d353. Can you have a look?
